Linking error with `Undefined symbols for architecture x86_64: _SCDynamicStoreCopyProxies` on some macOS environment

**Describe your environment** 
OS: macOS 10.15
XCode: 12
Bazel: bazelisk 1.9.0


**Steps to reproduce**

```bash
bazel --output_user_root=$HOME/.cache/bazel build --copt=-DENABLE_METRICS_PREVIEW --copt=-DENABLE_LOGS_PREVIEW --copt -Wno-unused-function --copt -Wno-error=unused-command-line-argument --host_linkopt -lpthread --host_linkopt -liconv --host_linkopt -lresolv --host_linkopt -ldl --host_linkopt -lm //...
```

**What is the expected behavior?**
Sompiling success

**What is the actual behavior?**

```bash
Use --sandbox_debug to see verbose messages from the sandbox
Undefined symbols for architecture x86_64:
  "_SCDynamicStoreCopyProxies", referenced from:
      _Curl_resolv in libcurl.a(hostip.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error in child process '/usr/bin/xcrun'. 1
```

**Additional context**
Should link with `-framework SystemConfiguration` on macOS.

--- a/bazel/curl.BUILD
+++ b/bazel/curl.BUILD
@@ -31,7 +31,6 @@ cc_library(
     ]),
     copts = CURL_COPTS + [
         "-DOS=\"os\"",
-        # "-DCURL_EXTERN_SYMBOL=__attribute__((__visibility__(\"default\")))",
     ],
     defines = ["CURL_STATICLIB"],
     includes = [

--- a/bazel/curl.BUILD
+++ b/bazel/curl.BUILD
@@ -6,13 +6,17 @@ package(features = ["no_copts_tokenization"])
 
 config_setting(
     name = "windows",
-    values = {"cpu": "x64_windows"},
+    constraint_values = [
+        "@bazel_tools//platforms:windows",
+    ],
     visibility = ["//visibility:private"],
 )
 
 config_setting(
     name = "osx",
-    values = {"cpu": "darwin"},
+    constraint_values = [
+        "@bazel_tools//platforms:osx",
+    ],
     visibility = ["//visibility:private"],
 )
 
@@ -27,11 +31,27 @@ cc_library(
     ]),
     copts = CURL_COPTS + [
         "-DOS=\"os\"",
-        "-DCURL_EXTERN_SYMBOL=__attribute__((__visibility__(\"default\")))",
+        # "-DCURL_EXTERN_SYMBOL=__attribute__((__visibility__(\"default\")))",
     ],
+    defines = ["CURL_STATICLIB"],
     includes = [
         "include/",
         "lib/",
     ],
+    linkopts = select({
+        "//:windows": [
+            "-DEFAULTLIB:ws2_32.lib",
+            "-DEFAULTLIB:advapi32.lib",
+            "-DEFAULTLIB:crypt32.lib",
+            "-DEFAULTLIB:Normaliz.lib",
+        ],
+        "//:osx": [
+            "-framework SystemConfiguration",
+            "-lpthread",
+        ],
+        "//conditions:default": [
+            "-lpthread",
+        ],
+    }),
     visibility = ["//visibility:public"],
 )

--- a/examples/http/BUILD
+++ b/examples/http/BUILD
@@ -4,9 +4,7 @@ cc_binary(
         "client.cc",
         "tracer_common.h",
     ],
-    # TODO: Move copts/linkopts for static CURL usage into shared bzl file.
     copts = [
-        # "-DCURL_STATICLIB",
         "-DWITH_CURL",
     ],
     linkopts = select({

--- a/examples/http/BUILD
+++ b/examples/http/BUILD
@@ -6,7 +6,7 @@ cc_binary(
     ],
     # TODO: Move copts/linkopts for static CURL usage into shared bzl file.
     copts = [
-        "-DCURL_STATICLIB",
+        # "-DCURL_STATICLIB",
         "-DWITH_CURL",
     ],
     linkopts = select({

--- a/exporters/elasticsearch/BUILD
+++ b/exporters/elasticsearch/BUILD
@@ -9,9 +9,9 @@ cc_library(
         "include/opentelemetry/exporters/elasticsearch/es_log_exporter.h",
         "include/opentelemetry/exporters/elasticsearch/es_log_recordable.h",
     ],
-    copts = [
-        "-DCURL_STATICLIB",
-    ],
+    # copts = [
+    #     "-DCURL_STATICLIB",
+    # ],
     linkopts = select({
         "//bazel:windows": [
             "-DEFAULTLIB:advapi32.lib",

--- a/exporters/elasticsearch/BUILD
+++ b/exporters/elasticsearch/BUILD
@@ -9,9 +9,6 @@ cc_library(
         "include/opentelemetry/exporters/elasticsearch/es_log_exporter.h",
         "include/opentelemetry/exporters/elasticsearch/es_log_recordable.h",
     ],
-    # copts = [
-    #     "-DCURL_STATICLIB",
-    # ],
     linkopts = select({
         "//bazel:windows": [
             "-DEFAULTLIB:advapi32.lib",

--- a/ext/src/http/client/curl/BUILD
+++ b/ext/src/http/client/curl/BUILD
@@ -8,7 +8,7 @@ cc_library(
     ],
     # TODO: Move copts/linkopts for static CURL usage into shared bzl file.
     copts = [
-        "-DCURL_STATICLIB",
+        # "-DCURL_STATICLIB",
         "-DWITH_CURL",
     ],
     include_prefix = "src/http/client/curl",

--- a/ext/src/http/client/curl/BUILD
+++ b/ext/src/http/client/curl/BUILD
@@ -6,9 +6,7 @@ cc_library(
         "http_client_curl.cc",
         "http_client_factory_curl.cc",
     ],
-    # TODO: Move copts/linkopts for static CURL usage into shared bzl file.
     copts = [
-        # "-DCURL_STATICLIB",
         "-DWITH_CURL",
     ],
     include_prefix = "src/http/client/curl",

--- a/ext/test/w3c_tracecontext_test/BUILD
+++ b/ext/test/w3c_tracecontext_test/BUILD
@@ -3,10 +3,6 @@ cc_binary(
     srcs = [
         "main.cc",
     ],
-    # TODO: Move copts/linkopts for static CURL usage into shared bzl file.
-    # copts = [
-    #     "-DCURL_STATICLIB",
-    # ],
     linkopts = select({
         "//bazel:windows": [
             "-DEFAULTLIB:advapi32.lib",

--- a/ext/test/w3c_tracecontext_test/BUILD
+++ b/ext/test/w3c_tracecontext_test/BUILD
@@ -4,9 +4,9 @@ cc_binary(
         "main.cc",
     ],
     # TODO: Move copts/linkopts for static CURL usage into shared bzl file.
-    copts = [
-        "-DCURL_STATICLIB",
-    ],
+    # copts = [
+    #     "-DCURL_STATICLIB",
+    # ],
     linkopts = select({
         "//bazel:windows": [
             "-DEFAULTLIB:advapi32.lib",

